src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -297,6 +296,31 @@ private void DisposeTokenSource(DisconnectedCircuitEntry entry)
             }
         }
 
+        internal ValueTask Terminate(string circuitId)
+        {
+            CircuitHost circuitHost;
+            DisconnectedCircuitEntry entry = default;
+            lock (CircuitRegistryLock)
+            {
+                if (ConnectedCircuits.TryGetValue(circuitId, out circuitHost) || DisconnectedCircuits.TryGetValue(circuitId, out entry))
+                {
+                    PermanentDisconnect(circuitHost ?? entry.CircuitHost);
+                }
+                else
+                {
+                    return default;
+                }
+            }
+            if (circuitHost != null)
+            {
+                return circuitHost.DisposeAsync();
+            }
+            else
+            {
+                return default;
+            }
+        }
```

Comment:
This logic (line 309-321) seems like it could be simplified to a single `if (circuitHost != null)` or `circuitHost?.DisposeAsync() ?? default`.